sibelius/relay-workshop
# feat(wp): try to upgrade to wp 5

How to test

`yarn server` -> should start server using webpack
`yarn web` -> should start web using webpack
`yarn update-schema` -> should update schema script using webpack

- [ ] reload-server-webpack-plugin, had to inline the plugin to fix tap warning
- [ ] @pmmmwh/react-refresh-webpack-plugin throws

```jsx
Uncaught exception: Error: NormalModuleFactory.afterResolve is no longer a waterfall hook, but a bailing hook instead. Do not return the passed object, but modify it instead. Returning false will ignore the request and results in no module created.
  [webpack-cli] Error: NormalModuleFactory.afterResolve is no longer a waterfall hook, but a bailing hook instead. Do not return the passed object, but modify it instead. Returning false will ignore the request and results in no module created.
  at relay-workshop/node_modules/webpack-cli/node_modules/webpack/lib/NormalModuleFactory.js:251:14
at Hook.eval [as callAsync] (eval at create (relay-workshop/node_modules/webpack-cli/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:14:1)
  at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (relay-workshop/node_modules/webpack-cli/node_modules/tapable/lib/Hook.js:18:14)
  at relay-workshop/node_modules/webpack-cli/node_modules/webpack/lib/NormalModuleFactory.js:247:30
  at eval (eval at create (relay-workshop/node_modules/webpack-cli/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at relay-workshop/node_modules/webpack-cli/node_modules/webpack/lib/NormalModuleFactory.js:481:7
    at relay-workshop/node_modules/webpack-cli/node_modules/webpack/lib/NormalModuleFactory.js:116:11
    at relay-workshop/node_modules/webpack-cli/node_modules/webpack/lib/NormalModuleFactory.js:513:8
    at NormalModuleFactory.resolveRequestArray (relay-workshop/node_modules/webpack-cli/node_modules/webpack/lib/NormalModuleFactory.js:734:34)
    at relay-workshop/node_modules/webpack-cli/node_modules/webpack/lib/NormalModuleFactory.js:505:11
```

- [ ] html-webpack-plugin is throwing

```jsx
Html Webpack Plugin:
  TypeError: Cannot read property 'tap' of undefined
  
  - ExternalModuleFactoryPlugin.js:17 ExternalModuleFactoryPlugin.apply
    [relay-workshop]/[webpack]/lib/ExternalModuleFactoryPlugin.js:17:37
  
  - ExternalsPlugin.js:16 
    [relay-workshop]/[webpack]/lib/ExternalsPlugin.js:16:63
  
  
  - Hook.js:14 Hook.CALL_DELEGATE [as _call]
    [relay-workshop]/[webpack-cli]/[tapable]/lib/Hook.js:14:14
  
  - Compiler.js:982 
    [relay-workshop]/[webpack-cli]/[webpack]/lib/Compiler.js:982:23
  
  
  - Hook.js:18 Hook.CALL_ASYNC_DELEGATE [as _callAsync]
    [relay-workshop]/[webpack-cli]/[tapable]/lib/Hook.js:18:14
  
  - Compiler.js:979 Compiler.compile
    [relay-workshop]/[webpack-cli]/[webpack]/lib/Compiler.js:979:28
  
  - Compiler.js:494 Compiler.runAsChild
    [relay-workshop]/[webpack-cli]/[webpack]/lib/Compiler.js:494:8
  
  - child-compiler.js:110 
    [relay-workshop]/[html-webpack-plugin]/lib/child-compiler.js:110:21
```
